server/src/main/java/org/elasticsearch/cluster/routing/IndexShardRoutingTable.java

```diff
@@ -242,38 +241,23 @@ public ShardIterator activeInitializingShardsRankedIt(
         return new PlainShardIterator(shardId, ordered);
     }
 
-    private static Set<String> getAllNodeIds(final List<ShardRouting> shards) {
-        final Set<String> nodeIds = new HashSet<>();
-        for (ShardRouting shard : shards) {
-            nodeIds.add(shard.currentNodeId());
-        }
-        return nodeIds;
-    }
-
-    private static Map<String, Optional<ResponseCollectorService.ComputedNodeStats>> getNodeStats(
-        final Set<String> nodeIds,
+    private static Map<String, ResponseCollectorService.ComputedNodeStats> getNodeStats(
+        final List<ShardRouting> shards,
         final ResponseCollectorService collector
     ) {
-
-        final Map<String, Optional<ResponseCollectorService.ComputedNodeStats>> nodeStats = Maps.newMapWithExpectedSize(nodeIds.size());
-        for (String nodeId : nodeIds) {
-            nodeStats.put(nodeId, collector.getNodeStatistics(nodeId));
+        final Map<String, ResponseCollectorService.ComputedNodeStats> nodeStats = new HashMap<>();
```

Comment:
Missing `Maps.newMapWithExpectedSize` here

Reply:
Would be nice to have that, but I have no clue what to pre-size for here? We could adjust the data-structures to be able to pre-size maybe, but right now I left it as it was in `getAllNodeIds` where we didn't pre-size the hashset to have a clear win.

Reply:
Ah, sorry, I was looking at the removed line below which does have a size. This one is doing something different to that